viewing_party/party.py

```diff
@@ -1,23 +1,130 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if not title or not genre or not rating:
+        return None
+    movie_dict = {
+        "title": title,
+        "genre": genre,
+        "rating": rating,
+    }
+    
+    return movie_dict
+
+def add_to_watched(user_data, movie):
+    if not movie:
+        return user_data
+    user_data["watched"].append(movie)
+    
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    if not movie:
+        return user_data
+    user_data["watchlist"].append(movie)
+    
+    return user_data
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if title == movie["title"]:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+            break
+    
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    if not user_data["watched"]:
+        return 0.0
+    
+    total_rating = 0
+    for movie in user_data["watched"]:
+        total_rating += movie["rating"]
+    return total_rating / len(user_data["watched"])
 
+def get_most_watched_genre(user_data):
+    if not user_data["watched"]:
+        return None
+    
+    genre_count = {}
+    for movie in user_data["watched"]:
+        genre = movie["genre"]
+        if genre in genre_count:
+            genre_count[genre] += 1
+        else:
+            genre_count[genre] = 1
+            
+    max_count = 0
+    most_watched_genre = None
+    for genre, count in genre_count.items():
+        if count > max_count:
+            max_count = count
+            most_watched_genre = genre
+    return most_watched_genre
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(user_data):
+    friends_watched_movie_titles = set()
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friends_watched_movie_titles.add(movie["title"])
+            
+    user_unique_watched = []
+    for movie in user_data["watched"]:
+        if movie["title"] not in friends_watched_movie_titles:
+            user_unique_watched.append(movie)
+    return user_unique_watched

```

Comment:
I love the way y'all implemented this! It is short, sweet and easy to read! The use of sets is great here and you've done a good job of optimizing the complexity!